src/services/rfqm_service.ts

```diff
@@ -0,0 +1,95 @@
+    amount: BigNumber; // in terms of the amount the taker is selling
+    buyToken: string;
+    sellToken: string;
+    takerAddress?: string;
+}
+
+export interface FetchIndicativeQuoteResponse {
+    buyAmount: BigNumber;
+    buyTokenAddress: string;
+    price: BigNumber;
+    sellAmount: BigNumber;
+    sellTokenAddress: string;
+}
+
+/**
+ * RfqmService is the coordination layer for HTTP based RFQM flows
+ */
+export class RfqmService {
+    constructor(private readonly _quoteRequestor: QuoteRequestor) {}
+
+    /**
+     * Fetch the best indicative quote available.
+     */
+    public async fetchIndicativeQuoteAsync(params: FetchIndicativeQuoteParams): Promise<FetchIndicativeQuoteResponse> {
+        const { amount, sellToken, buyToken, apiKey } = params;
+
+        // Map params to the terminology of Quote Requestor
+        const marketOperation = MarketOperation.Sell;
+        const assetFillAmount = amount;
+        const makerToken = buyToken;
+        const takerToken = sellToken;
+
+        // Fetch quotes
+        const opts = {
+            takerAddress: NULL_ADDRESS,
+            txOrigin: NULL_ADDRESS, // TODO - set to worker registry
+            apiKey,
+            intentOnFilling: false, // TODO - safe to hardcode?
+            isIndicative: true,
+            makerEndpointMaxResponseTimeMs: RFQT_REQUEST_MAX_RESPONSE_MS,
+            nativeExclusivelyRFQ: true,
+            altRfqAssetOfferings: {}, // TODO - set this to the alt rfq asset offerings
+            isLastLook: true,
+        };
+        const indicativeQuotes = await this._quoteRequestor.requestRfqmIndicativeQuotesAsync(
+            makerToken,
+            takerToken,
+            assetFillAmount,
+            marketOperation,
+            undefined,
+            opts,
+        );
+
+        // Filter out quotes that:
+        // - are for the wrong pair
+        // - cannot fill 100 % of the requested amount
+        //
+        // And sort by best price
+        const sortedQuotes = indicativeQuotes
+            .filter((q) => q.takerToken === takerToken && q.makerToken === makerToken)
+            .filter((q) => q.takerAmount.gte(assetFillAmount))
+            .sort((a, b) => {
+                const aPrice = a.makerAmount.div(a.takerAmount);
+                const bPrice = b.makerAmount.div(b.takerAmount);
+                return bPrice.minus(aPrice).toNumber();
+            });
+
+        // No quotes found
+        if (sortedQuotes.length === 0) {
+            return Promise.reject(new Error('No valid quotes'));
+        }
+
+        const bestQuote = sortedQuotes[0];
+
+        // Prepare response
+        // TODO: handle decimals properly in price
+        return {
+            buyAmount: bestQuote.makerAmount,
+            buyTokenAddress: bestQuote.makerToken,
+            sellAmount: bestQuote.takerAmount,
+            sellTokenAddress: bestQuote.takerToken,
+            price: bestQuote.makerAmount.div(bestQuote.takerAmount),
```

Comment:
It would be really ideal if we could return a similar output to the `price` endpoint for RFQT. Something like

```
{
    "price": "0.000433262422",
    "gas": "160600",
    "buyTokenAddress": "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48",
    "buyAmount": "5000000000",
    "sellTokenAddress": "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2",
    "sellAmount": "2166312110000000000",
    "allowanceTarget": "0xdef1c0ded9bec7f1a1670819833240f027b25eff",
}
```